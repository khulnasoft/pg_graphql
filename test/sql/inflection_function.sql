begin;
    create table account (
        id int primary key
    );

    create function _full_name(rec public.account)
        returns text
        immutable
        strict
        language sql
    as $$
        select 'Foo';
    $$;

    -- Inflection off, Overrides: off
    comment on schema public is e'@graphql({"inflect_names": false})';
    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "account") {
            fields {
                name
            }
          }
        }
        $$)
    );

    savepoint a;

    -- Inflection off, Overrides: on
    comment on function public._full_name(public.account) is E'@graphql({"name": "wholeName"})';
    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "account") {
            fields {
                name
            }
          }
        }
        $$)
    );

    rollback to savepoint a;

    -- Inflection on, Overrides: off
    comment on schema public is e'@graphql({"inflect_names": true})';
    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Account") {
            fields {
                name
            }
          }
        }
        $$)
    );

    -- Inflection on, Overrides: on
    comment on function public._full_name(public.account) is E'@graphql({"name": "WholeName"})';
    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Account") {
            fields {
                name
            }
          }
        }
        $$)
    );

rollback;
