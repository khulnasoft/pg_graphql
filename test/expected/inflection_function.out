begin;
    create table account (
        id int primary key
    );
    create function _full_name(rec public.account)
        returns text
        immutable
        strict
        language sql
    as $$
        select 'Foo';
    $$;
    -- Inflection off, Overrides: off
    comment on schema public is e'@graphql({"inflect_names": false})';
    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "account") {
            fields {
                name
            }
          }
        }
        $$)
    );
              jsonb_pretty               
-----------------------------------------
 {                                      +
     "data": {                          +
         "__type": {                    +
             "fields": [                +
                 {                      +
                     "name": "nodeId"   +
                 },                     +
                 {                      +
                     "name": "id"       +
                 },                     +
                 {                      +
                     "name": "full_name"+
                 }                      +
             ]                          +
         }                              +
     }                                  +
 }
(1 row)

    savepoint a;
    -- Inflection off, Overrides: on
    comment on function public._full_name(public.account) is E'@graphql({"name": "wholeName"})';
    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "account") {
            fields {
                name
            }
          }
        }
        $$)
    );
              jsonb_pretty               
-----------------------------------------
 {                                      +
     "data": {                          +
         "__type": {                    +
             "fields": [                +
                 {                      +
                     "name": "nodeId"   +
                 },                     +
                 {                      +
                     "name": "id"       +
                 },                     +
                 {                      +
                     "name": "wholeName"+
                 }                      +
             ]                          +
         }                              +
     }                                  +
 }
(1 row)

    rollback to savepoint a;
    -- Inflection on, Overrides: off
    comment on schema public is e'@graphql({"inflect_names": true})';
    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Account") {
            fields {
                name
            }
          }
        }
        $$)
    );
              jsonb_pretty              
----------------------------------------
 {                                     +
     "data": {                         +
         "__type": {                   +
             "fields": [               +
                 {                     +
                     "name": "nodeId"  +
                 },                    +
                 {                     +
                     "name": "id"      +
                 },                    +
                 {                     +
                     "name": "fullName"+
                 }                     +
             ]                         +
         }                             +
     }                                 +
 }
(1 row)

    -- Inflection on, Overrides: on
    comment on function public._full_name(public.account) is E'@graphql({"name": "WholeName"})';
    select jsonb_pretty(
        graphql.resolve($$
        {
          __type(name: "Account") {
            fields {
                name
            }
          }
        }
        $$)
    );
              jsonb_pretty               
-----------------------------------------
 {                                      +
     "data": {                          +
         "__type": {                    +
             "fields": [                +
                 {                      +
                     "name": "nodeId"   +
                 },                     +
                 {                      +
                     "name": "id"       +
                 },                     +
                 {                      +
                     "name": "WholeName"+
                 }                      +
             ]                          +
         }                              +
     }                                  +
 }
(1 row)

rollback;
